### PR TITLE
lib: bin: lwm2m_carrier: update to version 0.30.0

### PR DESCRIPTION
**Pull request by nordic-hani**
* Added latest version v0.30.0

Signed-off-by: Håvard Vermeer <havard.vermeer@nordicsemi.no>

### CODE REVIEW

**Comment by umapraseeda** [comment]
@nordic-hani, The doc updates to nrf_modem_lib.rst and the changelog (release-notes-changelog.rst) are already covered as part of https://github.com/nrfconnect/sdk-nrf/pull/7571/files. Those updates can be removed from this PR.

File: doc/nrf/libraries/bin/lwm2m_carrier/CHANGELOG.rst [hunk 1]
```diff
@@ -9,6 +9,97 @@ Changelog
...
+
++-------------------------+---------------+------------+
+|                         | Flash (Bytes) | RAM (Bytes)|
++-------------------------+---------------+------------+
+| Library size            |               |            |
+| (binary)                |               |            |
++-------------------------+---------------+------------+
+| Library size            |               |            |
+| (reference application) |               |            |
++-------------------------+---------------+------------+
```
Comment (umapraseeda) [inline doc/nrf/libraries/bin/lwm2m_carrier/CHANGELOG.rst:35]: Values need to be added to the table.

File: doc/nrf/libraries/bin/lwm2m_carrier/CHANGELOG.rst [hunk 2]
```diff
@@ -9,6 +9,97 @@ Changelog
...
+
+* Removed the Kconfig options ``CONFIG_LWM2M_CARRIER_USE_CUSTOM_PSK`` and ``CONFIG_LWM2M_CARRIER_USE_CUSTOM_APN``.
+
+  * Instead, only the Kconfig options :kconfig:option:`CONFIG_LWM2M_CARRIER_CUSTOM_PSK` and :kconfig:option:`CONFIG_LWM2M_CARRIER_CUSTOM_APN` are needed. If they are empty, they are ignored.
+
+* Renamed the event ``LWM2M_CARRIER_EVENT_CARRIER_INIT`` to :c:macro:`LWM2M_CARRIER_EVENT_INIT`.
+* Removed the event ``LWM2M_CARRIER_EVENT_CERTS_INIT`` and initialization parameter ``lwm2m_carrier_event_certs_init_t``.
+
+ * Instead, certificates can be written to modem when the :c:macro:`LWM2M_CARRIER_EVENT_INIT` even is received, before attaching to the network.
+ * List of certificates is no longer supplied to the :c:func:`l2m2m_carrier_init` function. LwM2M carrier library will instead iterate through all CA certificates in the modem.
```
Comment (umapraseeda) [inline doc/nrf/libraries/bin/lwm2m_carrier/CHANGELOG.rst:60]:
```suggestion
 * List of certificates is no longer supplied to the :c:func:`lwm2m_carrier_init` function. LwM2M carrier library will instead iterate through all CA certificates in the modem.
```

File: doc/nrf/libraries/bin/lwm2m_carrier/CHANGELOG.rst [hunk 3]
```diff
@@ -9,6 +9,97 @@ Changelog
...
+* Added the Kconfig option :kconfig:option:`CONFIG_LWM2M_CARRIER_SESSION_IDLE_TIMEOUT`.
+* Removed some runtime resource ``_set()`` functions. The resources are static and therefore added to library initialization instead.
+
+  * Removed ``lwm2m_carrier_device_type_set()``, ``lwm2m_carrier_hardware_version_set()`` and ``lwm2m_carrier_software_version_set()``.
+  * Added :kconfig:option:`CONFIG_LWM2M_CARRIER_DEVICE_TYPE`, :kconfig:option:`CONFIG_LWM2M_CARRIER_DEVICE_HARDWARE_VERSION` and :kconfig:option:`CONFIG_LWM2M_CARRIER_DEVICE_SOFTWARE_VERSION`.
+
+* Added new initialization configurations to set the static device object resources:
+
+  * :kconfig:option:`CONFIG_LWM2M_CARRIER_DEVICE_MANUFACTURER`.
+  * :kconfig:option:`CONFIG_LWM2M_CARRIER_DEVICE_MODEL_NUMBER`.
```
Comment (umapraseeda) [inline doc/nrf/libraries/bin/lwm2m_carrier/CHANGELOG.rst:71]:
```suggestion
  * :kconfig:option:`CONFIG_LWM2M_CARRIER_DEVICE_MANUFACTURER`
  * :kconfig:option:`CONFIG_LWM2M_CARRIER_DEVICE_MODEL_NUMBER`
```

File: doc/nrf/libraries/bin/lwm2m_carrier/app_integration.rst [hunk 4]
```diff
@@ -84,6 +107,16 @@ You can provide the initialization parameter :c:type:`lwm2m_carrier_config_t` to
 
 Note that these settings can put the LwM2M carrier library either in the normal mode where it connects to the applicable carriers, or in the generic mode where it can connect to any bootstrap server.
 
+* :kconfig:option:`CONFIG_LWM2M_CARRIER_DEVICE_MANUFACTURER`, :kconfig:option:`CONFIG_LWM2M_CARRIER_DEVICE_MODEL_NUMBER`, :kconfig:option:`CONFIG_LWM2M_CARRIER_DEVICE_TYPE`, :kconfig:option:`CONFIG_LWM2M_CARRIER_DEVICE_HARDWARE_VERSION`, :kconfig:option:`CONFIG_LWM2M_CARRIER_DEVICE_SOFTWARE_VERSION`:
+
+  * These values are reported in the Device Object. They are not expected to change during run time.
+  * If these values are not set by the application, the LwM2M carrier library sets the default values based on the modem and operator network.
+
+* :kconfig:option:`CONFIG_LWM2M_CARRIER_LG_UPLUS`, :kconfig:option:`CONFIG_LWM2M_CARRIER_LG_UPLUS_SERVICE_CODE`
```
Comment (umapraseeda) [inline doc/nrf/libraries/bin/lwm2m_carrier/app_integration.rst:115]:
```suggestion
* :kconfig:option:`CONFIG_LWM2M_CARRIER_LG_UPLUS`, :kconfig:option:`CONFIG_LWM2M_CARRIER_LG_UPLUS_SERVICE_CODE`:
```

**Comment by 57300** [comment]
`requirements.rst` has a list of AT notifications used by the library. It should also mention `AT+ODISNTF` and `AT%XSIM`.

File: applications/asset_tracker_v2/doc/asset_tracker_v2_description.rst [hunk 5]
```diff
@@ -180,6 +180,20 @@ See the library's documentation for more information and configuration options.
 
 To enable the LwM2M carrier library, add the parameter ``-DOVERLAY_CONFIG=overlay-carrier.conf`` to your build command.
 
+The CA root certificates needed for modem FOTA, are not provisioned in the Asset Tracker sample.
```
Comment (umapraseeda) [inline applications/asset_tracker_v2/doc/asset_tracker_v2_description.rst:183]:
```suggestion
The CA root certificates that are needed for modem FOTA are not provisioned in the Asset Tracker v2 application.
```

File: applications/asset_tracker_v2/doc/asset_tracker_v2_description.rst [hunk 6]
```diff
@@ -180,6 +180,20 @@ See the library's documentation for more information and configuration options.
...
+You can flash the :ref:`lwm2m_carrier` sample to write the certificates to modem before flashing the Asset Tracker,
+or use the at_client similar to what is shown in ref`lwm2m_client_provisioning`.
```
Comment (umapraseeda) [inline applications/asset_tracker_v2/doc/asset_tracker_v2_description.rst:185]:
```suggestion
You can flash the :ref:`lwm2m_carrier` sample to write the certificates to modem before flashing the Asset Tracker v2 application, or use the :ref:`at_client_sample` sample as explained in :ref:`Preparing the nRF9160: LwM2M Client sample for production <lwm2m_client_provisioning>`.
```

File: applications/asset_tracker_v2/doc/asset_tracker_v2_description.rst [hunk 7]
```diff
@@ -180,6 +180,20 @@ See the library's documentation for more information and configuration options.
...
+It is also possible to modify the asset tracker project itself to include the certificate provisioning, as demonstrated in the :ref:`lwm2m_carrier` sample.
```
Comment (umapraseeda) [inline applications/asset_tracker_v2/doc/asset_tracker_v2_description.rst:186]:
```suggestion
It is also possible to modify the Asset Tracker v2 project itself to include the certificate provisioning, as demonstrated in the :ref:`lwm2m_carrier` sample.
```

File: applications/asset_tracker_v2/overlay-carrier.conf [hunk 8]
```diff
@@ -38,3 +38,7 @@ CONFIG_MODEM_KEY_MGMT=y
 # Non-volatile Storage
 CONFIG_NVS=y
 CONFIG_MPU_ALLOW_FLASH_WRITE=y
+
+# Use SPM since TFM does not support hard-float
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/overlay-carrier.conf:42]: /cc @simensrostad have a look at this.

**Review by PFnord — APPROVED** [review]

**Review by Eoin-McMahon — APPROVED** [review]

File: doc/nrf/releases/release-notes-changelog.rst [hunk 9]
```diff
@@ -451,6 +451,18 @@ Libraries
...
+----------------
+
+* Updated:
+
+  * :ref:`liblwm2m_carrier_readme` library:
+
+    * Updated to v0.30.0.
+      See the :ref:`liblwm2m_carrier_changelog` for detailed information.
+    * Projects that use the LwM2M carrier library can not use TF-M for this release, since the LwM2M carrier library requires hard floating point.
+      for more information, please see the :ref:`ug_tfm <TF-M>` documentation.
```
[the remark this answers is not in the document]
Comment (umapraseeda) [reply doc/nrf/releases/release-notes-changelog.rst:464]: Please add this change to fix the doc build error.

**Comment by nordic-hani** [comment]
Regarding compliance `16206: WARNING:NEW_TYPEDEFS`: this has earlier been agreed upon and accepted for the LwM2M carrier library. Please contact me directly if there is a need for discussion.

**Comment by umapraseeda** [comment]
Regarding compliance ``#1101: FILE: doc/nrf/libraries/bin/lwm2m_carrier/images/lwm2m_carrier_msc_bootstrap.svg:107`` (https://github.com/nrfconnect/sdk-nrf/runs/6568297761?check_suite_focus=true#step:9:21), there is no spellcheck error (could be because of the length of the text). This can be ignored, correct @gmarull?

**Comment by gmarull** [comment]
> Regarding compliance `#1101: FILE: doc/nrf/libraries/bin/lwm2m_carrier/images/lwm2m_carrier_msc_bootstrap.svg:107` ([nrfconnect/sdk-nrf/runs/6568297761?check_suite_focus=true#step:9:21](https://github.com/nrfconnect/sdk-nrf/runs/6568297761?check_suite_focus=true#step:9:21)), there is no spellcheck error (could be because of the length of the text). This can be ignored, correct @gmarull?

For now, just delete the `<desc>...</desc>` entry in the SVG file, after all it's not the rendered text. We'll investigate if it is worth keeping SVG spell checks in CI.

**Review by umapraseeda — APPROVED** [review]

**Comment by umapraseeda** [comment]
> > Regarding compliance `#1101: FILE: doc/nrf/libraries/bin/lwm2m_carrier/images/lwm2m_carrier_msc_bootstrap.svg:107` ([nrfconnect/sdk-nrf/runs/6568297761?check_suite_focus=true#step:9:21](https://github.com/nrfconnect/sdk-nrf/runs/6568297761?check_suite_focus=true#step:9:21)), there is no spellcheck error (could be because of the length of the text). This can be ignored, correct @gmarull?
> 
> For now, just delete the `<desc>...</desc>` entry in the SVG file, after all it's not the rendered text. We'll investigate if it is worth keeping SVG spell checks in CI.

Discussed with @mirjaku. This line is being automatically added for all the components by default. So even if we remove it now, it might appear again as soon as we edit the visio source file next time and export the svg. I think either we ignore this compliance error for now and try to shorten the text in this particular block in future or disable compliance checks for svgs altogether after investigation.

**Review by divipillai — APPROVED** [review]

**Review by simensrostad — APPROVED** [review]
ATv2 changes looks fine.

**Review by lucasvdp — APPROVED** [review]

**Review by stig-bjorlykke — APPROVED** [review]

**Review by 57300 — APPROVED** [review]

**Pull request merged**